### PR TITLE
[COR-155] 'coredb install' to install operator

### PR DESCRIPTION
**Pull request by sjmiller609**
https://user-images.githubusercontent.com/7516283/213816129-dc8a6ec6-9871-4640-aa87-e2accff850f3.mov

### CODE REVIEW

File: .github/workflows/operator.yml [hunk 1]
```diff
@@ -85,6 +85,11 @@ jobs:
           prefix-key: "coredb-operator-functional-test"
           workspaces: |
             coredb-operator
+      - name: Check CRD is updated in the yaml directory
+        run: |
+          set -xe
+          cargo run --bin crdgen > yaml/crd.yaml
+          git diff --exit-code yaml/crd.yaml
```
Comment (sjmiller609) [inline .github/workflows/operator.yml:92]: this part will fail if CRD definition is changed but yaml directory not updated

File: .github/workflows/publish_operator.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# TODO: replace with push to CoreDB Quay, and versioned pushes.
+# This is to enable demo 'coredb install' on the CLI.
+#
+on:
+  push:
+    branches:
+      - main
+      - cli
```
Comment (sjmiller609) [inline .github/workflows/publish_operator.yml:8]: todo remove 'cli' after I check this works

File: coredb-operator/Dockerfile [hunk 3]
```diff
@@ -1,4 +1,17 @@
...
+
+FROM alpine:latest
+
+RUN adduser -D nonroot
+
+COPY --chown=nonroot:nonroot --from=builder target/x86_64-unknown-linux-musl/release/controller /app/controller
+COPY --chown=nonroot:nonroot --from=builder target/x86_64-unknown-linux-musl/release/crdgen /app/crdgen
+
+USER nonroot
+
```
Comment (sjmiller609) [inline coredb-operator/Dockerfile:15]: Although rust compiles to a binary, by default that binary has dependencies in the system like openssl and libc. If we build with MUSL we can avoid that - like clux did in the controller example.

File: coredb-operator/yaml/install.yaml [hunk 4]
```diff
@@ -86,7 +86,7 @@ spec:
       serviceAccountName: coredb-controller
       containers:
       - name: coredb-controller
-        image: localhost:5001/controller:ac593af6da990c911652c6e35fb777f36498d6dc
+        image: sjmiller609/coredb-operator:latest
```
Comment (sjmiller609) [inline coredb-operator/yaml/install.yaml:89]: replace this with quay when ian is back

**Review by ChuckHend — APPROVED** [review]

**Pull request merged**